test/Polar/hessian.jl

```diff
@@ -12,17 +14,22 @@ import ExaPF: PowerSystem, AutoDiff
 
 const PS = PowerSystem
 
-# Warning: currently works only on CPU, as depends on
-# explicit evaluation of Hessian, using MATPOWER expressions
 @testset "Compute reduced Hessian on CPU" begin
     @testset "Case $case" for case in ["case9.m", "case30.m"]
+        if has_cuda_gpu()
+            ITERATORS = zip([CPU(), CUDADevice()], [Vector, CuVector])
+        else
+            ITERATORS = zip([CPU()], [Vector])
+        end
+        @testset "Device $device" for (device, T) in ITERATORS
         ##################################################
         # Initialization
         ##################################################
         datafile = joinpath(dirname(@__FILE__), "..", "..", "data", case)
         tolerance = 1e-8
         pf = PS.PowerNetwork(datafile)
-        polar = PolarForm(pf, CPU())
+        polar = PolarForm(pf, device)
+        cpu_polar = PolarForm(pf, CPU())
```

Comment:
Oh, interesting! Why do we need a second `polar` on the CPU?

Reply:
Since we call the matpower hessians which can only be computed with a polar model on a CPU architecture. Maybe that's wrong.

Reply:
I see. I don't know what's best: either we compute Matpower Hessians on the CPU, or we don't compute them at all when we are testing on the GPU.

Reply:
I think in principle it's not bad to instantiate both. This makes also sure that there are no side effects.